packages/frontend/src/app/asset-risks/[address]/_components/clientside-logic.tsx

```diff
@@ -0,0 +1,26 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { type Address } from 'viem'
+import { api } from '~/trpc/react'
+
+export function ClientsideLogic({ address }: { address: Address }) {
+  const refreshTokens = api.assetRisks.refreshTokens.useMutation()
+  const refreshBalances = api.assetRisks.refreshBalances.useMutation()
+
+  const report = api.assetRisks.report.useQuery({ address })
+  const [refetched, setRefetched] = useState(false)
+
+  useEffect(() => {
+    if (refetched) return
+    setRefetched(true)
+    void (async () => {
+      await refreshTokens.mutateAsync({ address })
+      await report.refetch()
+      await refreshBalances.mutateAsync({ address })
+      await report.refetch()
```

Comment:
Why do we refetch the report twice here?

Reply:
Because these are two separate steps, ideally in the UI we'd display a list of tokens before we display the balances, it's not coded yet though (and it's good since we're changing the design)